ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -455,6 +481,9 @@ private void close(SelectionKey key) {
       activeConnections.set(this.keyMap.size());
       try {
         transmission.close();
+        if (pendingHandshakeTransmissions.get(transmission.getConnectionId()) != null) {
+          pendingHandshakeTransmissions.remove(transmission.getConnectionId());
+        }
       } catch (IOException e) {
```

Comment:
Also, I cannot think of why we would want to use this `Selector` across threads.